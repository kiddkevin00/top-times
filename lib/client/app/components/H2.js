import React, { Component } from 'react';
import cx from 'classnames';
import PropTypes from 'prop-types';

import '../../static/assets/styles/components/h2.scss';

// eslint-disable-next-line react/prefer-stateless-function
export default class H2 extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
    className: PropTypes.string,
  };

  static defaultProps = {
    className: '',
  };

  render() {
    const className = cx('pn-h2', this.props.className);

    return <h2 className={className}>{this.props.children}</h2>;
  }
}
