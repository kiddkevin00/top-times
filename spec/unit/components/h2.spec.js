import H2 from '../../../lib/client/app/components/H2';
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Header 2 component', () => {
  let props;

  beforeEach(() => {
    props = {
      className: 'test-class',
      children: 'This is a header 2',
    };
  });

  it('should be rendered correctly', () => {
    const wrapper = shallow(<H2 {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should always have predefined CSS classes', () => {
    const wrapper = shallow(<H2 {...props}>click me</H2>);

    expect(wrapper.find('.pn-h2').length).toBe(1);
  });
});
