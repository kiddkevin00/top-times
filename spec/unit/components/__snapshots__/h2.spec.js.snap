// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header 2 component should be rendered correctly 1`] = `
<h2
  className="pn-h2 test-class"
>
  This is a header 2
</h2>
`;
